src/WCSimWLSProperties.cc

```diff
@@ -0,0 +1,379 @@
+	2216.06, 1673.89, 1277.31, 920.492, 651.354, 451.711, 313.114, 221.159, 156.305, 112.801, 
+	84.9561, 63.2998, 49.6208, 39.5509, 33.5793, 29.6825, 29.0258, 22.8064, 23.0578, 26.614, 
+	28.7748, 26.0248, 25.8214, 27.3878, 27.049, 26.1555, 25.1733, 24.8057, 28.0867, 28.449, 
+	28.9368, 29.8021, 27.8966, 28.1537, 27.7476, 27.9881, 20.4094, 20.4175, 24.0545, 23.1038, 
+	22.9155, 22.0755, 21.7028, 23.7999, 21.7777, 16.2794, 13.9256, 17.4295, 15.5754, 16.5594, 
+	14.0908, 14.7582, 18.5307, 19.4294, 18.4887, 18.5516, 17.0382, 20.9751, 23.2415, 20.0518, 
+	23.8798, 24.7696, 26.9215, 29.9342, 27.0665, 31.2296, 31.5986, 33.4633, 35.2167, 32.4512, 
+	29.0157, 30.2677, 31.2907, 30.7118, 25.3401, 25.8986, 23.5157, 22.4214, 22.5397, 23.051, 
+	19.1204, 18.9691, 16.9159, 17.782, 13.2649, 13.6162, 19.0826, 19.6119, 20.0295, 18.2931, 
+	17.4625, 20.5512, 23.6618, 22.853, 19.2703, 18.6003, 19.4221, 19.7589, 21.723, 20.3116, 
+	19.8483, 20.4982, 19.8304, 20.1763, 18.5058, 19.8518, 20.2992, 21.4355, 16.6702, 14.5394, 
+	18.8193, 20.2057, 21.0858, 17.9655, 19.6285, 19.854, 22.0102, 22.379, 21.9284, 21.9469, 
+	20.9204, 21.8176, 21.4981, 17.3374, 20.12, 22.3727, 22.5763, 18.1656, 16.5643, 18.0199, 
+	19.8482, 19.7989, 18.6625, 18.8546, 20.3136, 21.7623, 20.3002, 19.292, 18.5301, 21.734, 
+	24.9841, 23.0722, 21.2413, 22.2118, 25.0339, 23.5018, 21.9702, 19.3274, 18.9602, 19.2639, 
+	21.6156, 14.8791, 17.5215, 16.4008, 18.6484, 18.0436, 13.564, 15.1429, 15.2968, 20.8932, 
+	20.8696, 21.6895, 21.518, 20.474, 27.0346, 22.812, 31.0448, 25.841, 29.9976, 25.1321, 
+	26.5766, 22.2891, 22.988, 25.2949, 19.6909, 27.5086, 22.1865, 28.2814, 29.5177, 32.6297
+      };
+
+  return eljen_component_vs_energy;
+}
+
+// ###################### //
+///////// STACKING /////////
+// ###################### //
+
+void EljenEJ286::SetgAbs(){
+  // double AbsAmp[nEntries_abslength_eljen] =
+  //     { 0, 0, 0, 0, 0, // 700 --- 650
+  //       0, 0, 0, 0, 0, // 650 --- 600
+  //       0, 0, 0, 0, 0, // 600 --- 550
+  //       0, 0, 0, 0, 0, // 550 --- 500
+  //       0, 0, 0, 0, 0, // 500 --- 450
+  //       0, 0, 0.01, 0.05, 0.64, // 450 --- 400
+  //       0.93, 0.9999, 0.9999, 0.9999, 0.9999, // 400 --- 350
+  //       0.9999, 0.9999, 0.99, 0.97, 0.89, // 350 --- 300
+  //       0.71, 0.54, 0, 0, // 300 --- 250
+  //       0, 0, 0, 0, 0}; // 250 --- 200
+  gAbs = new TGraph();
+  const int step = 10;
+
+  // std::cout << "Creating Stacking graph objects for WLS" << std::endl;
+  // std::cout << "WLS ABS" << std::endl;
+  // Then fill with point there is data
+  for(int i=nEntries_abslength_eljen-1; i>0; i--){
+    //    gAbs->SetPoint(i,round(hWL(GetPhotonEnergy_ABS()[i]))*nm,AbsAmp[i]);
+    gAbs->SetPoint(i,round(hWL(GetPhotonEnergy_ABS()[i]))*nm,1. - exp(-1.2*cm/GetAbs()[i]));
+    // std::cout << round(hWL(GetPhotonEnergy_ABS()[i]))*nm << "nm -> " << AbsAmp[i] << std::endl;
+  }
+
+}
+
+void EljenEJ286::SetgEm(){
+  gEm = new TGraph();
```

Comment:
It's more efficient to allocate the TGraph space required, rather than it having to do it multiple times as it's filled
```suggestion
  gEm = new TGraph(nEntries_eljen_component);
```
Also, please check if the graph exists, and delete it if it does

Reply:
Done, see cbd79d33abcdca16a32d19fd8ae3294921acd240